Modules/System/Rest Client/src/RestClient.Codeunit.al

```diff
@@ -0,0 +1,403 @@
+    /// <param name="RequestUri">The Uri the request is sent to.</param>
+    /// <returns>The response content as JsonToken</returns>
+    procedure GetAsJson(RequestUri: Text) JsonToken: JsonToken
+    var
+        HttpResponseMessage: Codeunit "Http Response Message";
+    begin
+        HttpResponseMessage := Send(Enum::"Http Method"::GET, RequestUri);
+        if not HttpResponseMessage.GetIsSuccessStatusCode() then
+            Error(HttpResponseMessage.GetErrorMessage());
+
+        JsonToken := HttpResponseMessage.GetContent().AsJson();
+    end;
+
+    /// <summary>Sends a POST request to the specified Uri and returns the response content as JsonToken.</summary>
+    /// <remarks>The function fails with an error message if the request could not be sent or a response was not received.
+    /// The function also fails in case the response does not contain a success status code or a valid JSON content.</remarks>
+    /// <param name="RequestUri">The Uri the request is sent to.</param>
+    /// <param name="Content">The content to send as a JsonObject.</param>
+    /// <returns>The response content as JsonToken</returns>
+    procedure PostAsJson(RequestUri: Text; Content: JsonObject) Response: JsonToken
+    begin
+        Response := PostAsJson(RequestUri, Content.AsToken());
+    end;
+
+    /// <summary>Sends a POST request to the specified Uri and returns the response content as JsonToken.</summary>
+    /// <remarks>The function fails with an error message if the request could not be sent or a response was not received.
+    /// The function also fails in case the response does not contain a success status code or a valid JSON content.</remarks>
+    /// <param name="RequestUri">The Uri the request is sent to.</param>
+    /// <param name="Content">The content to send as a JsonArray.</param>
+    /// <returns>The response content as JsonToken</returns>
+    procedure PostAsJson(RequestUri: Text; Content: JsonArray) Response: JsonToken
+    begin
+        Response := PostAsJson(RequestUri, Content.AsToken());
+    end;
+
+    /// <summary>Sends a POST request to the specified Uri and returns the response content as JsonToken.</summary>
+    /// <remarks>The function fails with an error message if the request could not be sent or a response was not received.
+    /// The function also fails in case the response does not contain a success status code or a valid JSON content.</remarks>
+    /// <param name="RequestUri">The Uri the request is sent to.</param>
+    /// <param name="Content">The content to send as a JsonToken.</param>
+    /// <returns>The response content as JsonToken</returns>
+    procedure PostAsJson(RequestUri: Text; Content: JsonToken) Response: JsonToken
+    var
+        HttpResponseMessage: Codeunit "Http Response Message";
+        HttpContent: Codeunit "Http Content";
+    begin
+        HttpResponseMessage := Send(Enum::"Http Method"::POST, RequestUri, HttpContent.Create(Content));
```

Comment:
seems like this should move to the Impl.

Reply:
same for the other instances below...

Reply:
Can you please elaborate? Should a Facade codeunit not use any of its own functions or overloads?

Reply:
A façade can use its own methods but shouldn't contain logic so really, when it calls its own methods its generally only calling another overload with some default value.

Reply:
Moved all logic to implementation codeunit.